manager/dispatcher/dispatcher_test.go

```diff
@@ -460,14 +461,20 @@ func TestAssignmentsInitialNodeTasks(t *testing.T) {
 	}
 
 	// updating all the tasks will attempt to remove all the secrets for the tasks that are in state > running
-	err = gd.Store.Update(func(tx store.Tx) error {
-		for _, task := range tasks {
-			assert.NoError(t, store.UpdateTask(tx, task))
-		}
-		return nil
-
-	})
-	assert.NoError(t, err)
+	assert.NoError(t, testutils.PollFuncWithTimeout(nil, func() error {
```

Comment:
The intention was to prevent a mismatched version number, I guess in case something updates one of the tasks in between the get and update.